test/playwright/visual-regression/components/header.spec.ts

```diff
@@ -2,65 +2,43 @@ import { test } from '@playwright/test'
 
 import breakpoints from '~~/test/playwright/utils/breakpoints'
 import { hideInputCursors } from '~~/test/playwright/utils/page'
+import { goToSearchTerm, renderDirs } from '~~/test/playwright/utils/navigation'
 
 const headerSelector = '.main-header'
 const loadMoreSelector = 'button:has-text("Load more")'
 
 test.describe('header snapshots', () => {
-  test.describe('ltr', () => {
-    test.beforeEach(async ({ page }) => {
-      await page.goto('/search/?q=birds')
-    })
-
-    test.describe('header', () => {
-      breakpoints.describeEvery(({ expectSnapshot }) => {
-        test('resting', async ({ page }) => {
-          await expectSnapshot('resting-ltr', page.locator(headerSelector))
-        })
-
-        test('scrolled', async ({ page }) => {
-          await page.locator(loadMoreSelector).focus()
-          await expectSnapshot('scrolled-ltr', page.locator(headerSelector))
-        })
-
-        test('searchbar hovered', async ({ page }) => {
-          await page.hover('input')
-          await hideInputCursors(page)
-          await expectSnapshot(
-            'searchbar-hovered-ltr',
-            page.locator(headerSelector)
-          )
-        })
+  for (const dir of renderDirs) {
+    test.describe(dir, () => {
```

Comment:
Given iterations over page directions should always be followed by `test.describe(dir, ...)`, should we just go ahead and make a function like the breakpoints have? `describeDirections((dir) => ...)`?

Reply:
Makes sense 👍

Reply:
Actually, I don't know how to implement this. The problem is that the direction is set by the URL, and not the Playwright settings as in `breakpoints`.

Reply:
Ahh, interesting. The `describeDirections` callback could be `(getPath: (path: `/${string}`) => `${'/ar' | ''}${path}`, dir: 'ltr' | 'rtl') => void`. Then we eliminate having to import the path with direction function too:

```ts
describeDirections((getPath, dir) => {
  test('should render whatever', async ({ page }) => {
    await page.goto(getPath('/'))
    expect(page.screenshot()).toMatchSnapshot({ name: `whatever-${dir}` })
  })
})
```

What do you think?

Reply:
I've tried implementing this, but I really don't understand how :( Can we create a separate issue for `describeDirections`, too, @sarayourfriend?

Reply:
Sure, no problem.